### ISSUE TITLE
Environment Variables aren't relocated to the Sandbox

### ISSUE THREAD

**Issue by rgrinberg** [ocaml/dune#10294]
When we run actions inside the sandbox, we set various environment variables. The variables we need to relocate are:

* `PATH`
* `OPAM_SWITCH_PREFIX`
* `OCAMLPATH`
* `OCAMLTOP_INCLUDE_PATH`
* `MANPATH`
* `CAML_LD_LIBRARY_PATH`
* `OCAMLFIND_IGNORE_DUPS_IN`

All the variables above are computed automatically by dune when computing the environment from the set of packages. Since we manage these variables, we should take care to relocate them.

We should be modifying such paths into paths in the sandbox.

cc @gridbugs

**Comment by gridbugs** [comment]
This sounds similar to https://github.com/ocaml/dune/pull/10267 so I'll work on fixing this too (in a separate PR).

**Comment by Alizter** [comment]
This should be easier to do after #14373.
